lib/pure/net.nim

```diff
@@ -63,7 +68,14 @@ runnableExamples("-r:off -d:ssl"):
 
 runnableExamples("-r:off"):
   let socket = newSocket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)
-  socket.sendTo("192.168.0.1", Port(27960), "status\n")
+  let bytes = socket.sendTo("192.168.0.1", Port(27960), "status\n")
+  echo("sent: ", bytes)
```

Comment:
```suggestion
  doAssert socket.sendTo("192.168.0.1", Port(27960), "status\c\l") == 8
```

Reply:
> While reviewing this I got a thought: couldn't the `recvFrom`/`connect` procs instead detect that the `string` is an IP address and skip the domain resolution? if this is trivial then we should do it this way.

It crossed my mind too, but for IPv4 it can be a bit difficult to tell if it's an IP address or just a weird domain. 

> I think if we do this we can still allow `IpAddress` to be passed, but maybe this can be done with a generic: `proc recvForm[T: IpAddress | string](..., address: var T, ...)` to reduce code duplication?

This was more of my first approach, and I think it's likely better. It's tricky for a couple of the proc's but doable. I wasn't sure if the generic was frowned up in the standard library. The help docs were easier then to say "will parse hostname, or if an IpAddress will use it directly".